### DISCUSSION TITLE
How can I bind a my own object as a global helper to the goja vm?

### DISCUSSION THREAD

**Discussion by mskullcap** [pocketbase/pocketbase#3599] [Q&A] [answered]
Apologies if this is a silly question, I am new to Go and I haven't been able to find or figure out the answer for this...

I would like to use DuckDb (https://duckdb.org/) inside pb_hooks.  I have added https://github.com/marcboeker/go-duckdb and I can run duckdb queries in my customized Pocketbase executable.  However, I would really love to be able to experiment with my database connection in javascript pb_hooks.  Is there a way I can bind my duckdb connection to the goja vm context so I can access the connection?

**Comment by ganigeorgiev** [comment, marked as answer]
The goja.Runtime factories are not exposed at the moment (_I guess it is possible to add a configuration option in the [`jsvm.Config`](https://github.com/pocketbase/pocketbase/blob/master/plugins/jsvm/jsvm.go#L41) struct but I don't fully understand the use case for it_).

As a workaround, from the Go code you can "register" your custom Go initialized value to the shared app cache and then access it from inside the js `pb_hooks` through the `$app` object.

```go
app.Cache().Set("duckdb", db)
```

```js
const db = $app.cache().get("duckdb")
```

Note that even with this, depending on what actually you want to do with duckdb, there is no guarantee that it'll be sufficient and extra bindings for specific duckdb structs may be needed.

**Reply by mskullcap** [reply]
Ah, perfect!  I was hunting around for a map where I could stuff my object... the cache will work.  Thank you for your quick answer - and your tireless support of this brilliant platform.
